src/components/BaseCheckbox/BaseCheckbox.tsx

```diff
@@ -0,0 +1,32 @@
+import computeClassNames from '../../utils/compute-class-names';
+import './BaseCheckbox.css';
+
+interface BaseCheckboxProps {
+  value: boolean;
+  onChange?: (value: boolean) => void;
+  label?: string;
+  fullWidth?: boolean;
+  className?: string;
+}
+
+function BaseCheckbox({ value = false, onChange, label = '', fullWidth = true, className = '' }: BaseCheckboxProps) {
+  const computedClassNames = computeClassNames({
+    'base-checkbox--checked': value,
+    'base-checkbox--fullWidth': fullWidth
```

Comment:
Seguir la nomemclatura de `kebab-case` para las clases
```suggestion
    'base-checkbox--full-width': fullWidth
```

Recuerda actualizar en el archivo css tambien.